pygfx/utils/transform.py

```diff
@@ -0,0 +1,449 @@
+
+    @property
+    def y(self) -> float:
+        return self.position[1]
+
+    @property
+    def z(self) -> float:
+        return self.position[2]
+
+    @property
+    def scale_x(self) -> float:
+        return self.scale[0]
+
+    @property
+    def scale_y(self) -> float:
+        return self.scale[1]
+
+    @property
+    def scale_z(self) -> float:
+        return self.scale[2]
+
+    @position.setter
+    def position(self, value):
+        self.matrix = la.matrix_make_transform(value, self.rotation, self.scale)
+
+    @rotation.setter
+    def rotation(self, value):
+        self.matrix = la.matrix_make_transform(self.position, value, self.scale)
+
+    @scale.setter
+    def scale(self, value):
+        self.matrix = la.matrix_make_transform(self.position, self.rotation, value)
+
+    @x.setter
+    def x(self, value):
+        _, y, z = self.position
+        self.position = (value, y, z)
+
+    @y.setter
+    def y(self, value):
+        x, _, z = self.position
+        self.position = (x, value, z)
+
+    @z.setter
+    def z(self, value):
+        x, y, _ = self.position
+        self.position = (x, y, value)
+
+    @scale_x.setter
+    def scale_x(self, value):
+        _, y, z = self.scale
+        self.scale = (value, y, z)
+
+    @scale_y.setter
+    def scale_y(self, value):
+        x, _, z = self.scale
+        self.scale = (x, value, z)
+
+    @scale_z.setter
+    def scale_z(self, value):
+        x, y, _ = self.scale
+        self.scale = (x, y, value)
+
+    def __array__(self, dtype=None):
+        return self.matrix.astype(dtype)
+
+
+class AffineTransform(AffineBase):
+    """An affine tranformation"""
+
+    def __init__(
+        self,
+        matrix=None,
+        /,
+        *,
+        position=None,
+        rotation=None,
+        scale=None,
+    ) -> None:
+        super().__init__()
+        self.last_modified = perf_counter_ns()
+
+        if matrix is None:
+            matrix = np.eye(4, dtype=float)
+
+        self.untracked_matrix = np.asarray(matrix)
+
+        if position is not None:
+            self.position = position
+
+        if rotation is not None:
+            self.rotation = rotation
+
+        if scale is not None:
+            self.scale = scale
+
+    def flag_update(self):
+        self.last_modified = perf_counter_ns()
+        super().flag_update()
+
+    @property
+    def matrix(self):
+        view_array = self.untracked_matrix.view()
+        view_array.flags.writeable = False
+        return view_array
+
+    @matrix.setter
+    def matrix(self, value):
+        self.untracked_matrix[:] = value
+        self.flag_update()
+
+    def __matmul__(self, other):
+        if isinstance(other, AffineTransform):
+            return AffineTransform(self.matrix @ other.matrix)
+
+        return np.asarray(self) @ other
+
+    def __array__(self, dtype=None):
+        return self.untracked_matrix.astype(dtype, copy=False)
+
+
+class ChainedTransform(AffineBase):
```

Comment:
I think ChainedTransform is not used anywhere anymore, so you can remove it?

Reply:
Yes, `ChainedTransform` is no longer in use. Kept it since I wrote it and wasn't sure if you guys find it valuable for something in the future. Happy to remove it :)